### ISSUE TITLE
Just another crash

### ISSUE THREAD

**Issue by kiler4747** [canton7/SyncTrayzor#114]
```
Имя журнала:   Application
Источник:      .NET Runtime
Дата:          17.06.2015 13:14:37
Код события:   1026
Категория задачи:Отсутствует
Уровень:       Ошибка
Ключевые слова:Классический
Пользователь:  Н/Д
Компьютер:     NAS-PC
Описание:
Приложение: SyncTrayzor.exe
Версия платформы: v4.0.30319
Описание. Процесс был завершен из-за необработанного исключения.
Сведения об исключении: System.Threading.Tasks.TaskCanceledException
Стек:
   в System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>b__5(System.Object)
   в System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   в System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   в System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   в System.Threading.ThreadPoolWorkQueue.Dispatch()

Xml события:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name=".NET Runtime" />
    <EventID Qualifiers="0">1026</EventID>
    <Level>2</Level>
    <Task>0</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2015-06-17T10:14:37.000000000Z" />
    <EventRecordID>30381</EventRecordID>
    <Channel>Application</Channel>
    <Computer>NAS-PC</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Приложение: SyncTrayzor.exe
Версия платформы: v4.0.30319
Описание. Процесс был завершен из-за необработанного исключения.
Сведения об исключении: System.Threading.Tasks.TaskCanceledException
Стек:
   в System.Runtime.CompilerServices.AsyncMethodBuilderCore.&lt;ThrowAsync&gt;b__5(System.Object)
   в System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   в System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   в System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   в System.Threading.ThreadPoolWorkQueue.Dispatch()
</Data>
  </EventData>
</Event>
```

**Comment by canton7** [comment]
Can you post the log as well please? What did you do to trigger the crash? Can you reproduce it? Thanks!

**Comment by kiler4747** [comment]
I can't tell what happed, when I come home it was already crashed
here log, nothing interesting
```
2015-06-17 13:14:40.8325 [Debug] SyncTrayzor.SyncThing.TransferHistory.SyncThingTransferHistory: Item finished. Folder: Programming, Item: test\ParallelLoadBD\ParallelLoadBD\App.xaml.cs, Type: File, Action: Update 
2015-06-17 13:14:40.8325 [Debug] SyncTrayzor.SyncThing.TransferHistory.SyncThingTransferHistory: File Transfer set to complete: <FileTransfer Folder=Programming Path=test\ParallelLoadBD\ParallelLoadBD\App.xaml.cs Status=Completed ItemType=File ActionType=Update Started=17.06.2015 10:14:40 Finished=17.06.2015 10:14:40> 
2015-06-17 13:14:40.8325 [Debug] SyncTrayzor.SyncThing.EventWatcher.SyncThingEventWatcher: <ItemStarted ID=36342 Time=17.06.2015 13:13:46 Item=VideoFileRenamer\Code\Desktop Client\VideoFileRenamer\.git\objects\e7\da4cea99ac61f153a01113485f97ebff1eed97 Folder=Programming Type=file Action=update> 
2015-06-17 13:14:40.8325 [Debug] SyncTrayzor.SyncThing.TransferHistory.SyncThingTransferHistory: Item started. Folder: Programming, Item: VideoFileRenamer\Code\Desktop Client\VideoFileRenamer\.git\objects\e7\da4cea99ac61f153a01113485f97ebff1eed97, Type: File, Action: Update 
2015-06-17 19:39:41.7939 [Debug] SyncTrayzor.Services.MemoryUsageLogger: Working Set: 156MiB. Private Memory Size: 113MiB. GC Total Memory: 41MiB 
```

**Comment by canton7** [comment]
Hmm, odd. I had a suspicion as to what it might be, but that doesn't match up to the log.

As an aside: from experience trying to synchronise git repos will definitely lead to large amounts of pain. Conflicts on refs, necessary objects not being present, etc, are a pain in the arse. Git has a perfectly fine way of synchronizing repositories: use it.

**Comment by kiler4747** [comment]
I use it just like backup on my nas

**Comment by canton7** [comment]
Do double-check the git repos regularly: I had them corrupt beyond repair. It's much better to push them somewhere, or bundle them then backup/sync the bundle

**Comment by kiler4747** [comment]
Thanks for advice, I'll add git folder to ignor

**Comment by canton7** [comment]
Anyway, I've fixed what I think the problem might have been, and a build will be available here in a few minutes: https://ci.appveyor.com/project/canton7/synctrayzor/build/1.0.22.54/artifacts

Of course, with no way to reproduce it's very hard to say whether it was actually fixed...

**Comment by canton7** [comment]
That fix is in 1.0.23. I'll close this off for now, let me know if it crops up again
